src/Cli/dotnet/commands/dotnet-test/TestApplication.cs

```diff
@@ -223,21 +224,63 @@ private void StoreOutputAndErrorData(Process process)
 
         private bool ModulePathExists()
         {
-            if (!File.Exists(_modulePath))
+            if (!File.Exists(_module.DLLPath))
             {
-                ErrorReceived.Invoke(this, new ErrorEventArgs { ErrorMessage = $"Test module '{_modulePath}' not found. Build the test application before or run 'dotnet test'." });
+                ErrorReceived.Invoke(this, new ErrorEventArgs { ErrorMessage = $"Test module '{_module.DLLPath}' not found. Build the test application before or run 'dotnet test'." });
                 return false;
             }
             return true;
         }
 
+        private string BuildArgsWithDotnetRun(BuiltInOptions builtInOptions)
+        {
+            StringBuilder builder = new();
+
+            builder.Append($"{CliConstants.DotnetRunCommand} {CliConstants.ProjectOptionKey} \"{_module.ProjectPath}\"");
+
+            if (builtInOptions.HasNoRestore || builtInOptions.HasNoBuild)
```

Comment:
I think we should remove `builtInOptions.HasNoBuild` from this to be in line with user command line.